addons/html_editor/static/src/others/dynamic_field/dynamic_field_plugin.js

```diff
@@ -0,0 +1,341 @@
+        this.fieldPopover = this.dependencies.overlay.createOverlay(FieldSelectorPopover, {
+            hasAutofocus: true,
+            editable: this.editable,
+            className: "bg-light",
+        });
+
+        this.addDomListener(
+            this.editable,
+            "click",
+            (ev) => {
+                if (ev.detail === 1) {
+                    this.setFullSelection(ev.srcElement);
+                }
+            },
+            true
+        );
+    }
+
+    isValidTargetForDomListener(ev) {
+        return (
+            (ev.type === "click" && ev.target && closestElement(ev.target, "[t-field]")) ||
+            super.isValidTargetForDomListener(ev)
+        );
+    }
+
+    updateDynamicModel(resModel) {
+        this.resModel = resModel;
+    }
+
+    getPopoverTarget(isEdit) {
+        if (isEdit) {
+            const elements = this.dependencies.selection
+                .getTargetedNodes()
+                .filter((n) => n.nodeType === Node.ELEMENT_NODE);
+
+            if (elements.length === 1 && elements[0].hasAttribute("t-field")) {
+                // For now, ban expressions that contain a parenthesis
+                // That would be a function call (usually sudo)
+                // but we are *really* not sure about this.
+                if (elements[0].getAttribute("t-field").includes("(")) {
+                    return;
+                }
+                return elements[0];
+            }
+        } else {
+            const { anchorNode } = this.dependencies.selection.getEditableSelection();
+            return anchorNode.nodeType === Node.ELEMENT_NODE
+                ? anchorNode
+                : anchorNode.parentElement;
+        }
+    }
+
+    async editField() {
+        const target = this.getPopoverTarget(true);
+        if (!target) {
+            return;
+        }
+
+        await this.config.dynamicFieldPreprocess?.({
+            resModel: this.resModel,
+            element: target,
+        });
+
+        const { resModel, fieldPath: initialPath } = getQwebContext(target, this.resModel);
+
+        const initialLabel =
+            target.innerText ||
+            target.getAttribute("data-oe-demo") ||
+            target.getAttribute("data-oe-expression-readable");
+
+        this.fieldPopover.open({
+            target,
+            props: {
+                resModel,
+                path: initialPath,
+                label: initialLabel,
+                filter: this.filter.bind(this),
+                close: () => this.fieldPopover.close(),
+                validate: async ({ path, label, fieldInfo }) => {
+                    if (path !== initialPath) {
+                        const fullPath = this.getFieldPath(target, path);
+                        await this.setFieldAttributes(target, path, fullPath, fieldInfo);
+
+                        if (this.config.dynamicFieldPostprocess) {
+                            await this.config.dynamicFieldPostprocess({
```

Comment:
check with nby/html_editor if resources *could* be used async

otherwise ok, next step refactor mail rpc